### ISSUE TITLE
'unrecognized heap2 record type 32' with PG 9.3

### ISSUE THREAD

**Issue by dev1ant** [vmware/pg_rewind#42]
I've caught such message from pg_rewind while rewinding old master of PostgreSQL 9.3.5. Here is the console output:
```
postgres@xivastore01e /tmp $ /usr/pgsql-9.3/bin/pg_rewind -v --target-pgdata=/var/lib/pgsql/9.3/data/ --source-server='host=xivastore01d.mail.yandex.net port=5432 dbname=postgres connect_timeout=1'
connected to remote server
fetched file "global/pg_control", length 8192
fetched file "pg_xlog/00000004.history", length 135
The servers diverged at WAL position 1AC7/57F5F3E8 on timeline 3.
Rewinding from Last common checkpoint at 1AC6/C3565398 on timeline 3
unrecognized heap2 record type 32
postgres@xivastore01e /tmp $ /usr/pgsql-9.3/bin/pg_xlogdump -t 3 -s 1AC6/C3565398 -e 1AC7/57F5F3E8 >/tmp/xlogdump
postgres@xivastore01e /tmp $
```

Compressed output of pg_xlogdump could be taken [here](http://simply.name/tmp/xlogdump.tar.gz).

**Comment by michaelpq** [comment]
I think that's because we are simply missing XLOG_HEAP2_FREEZE_PAGE in parsexlog.c. I will draft a fix.

**Comment by michaelpq** [comment]
Pushed a fix as 84e6c80.

**Comment by dev1ant** [comment]
Thanks, Michael. It took me some time to reproduce this type of record in WAL but I have checked that it works as expected now.
